db/view/view_building_worker.cc

```diff
@@ -318,6 +318,28 @@ std::vector<service::view_building::view_building_task> view_building_worker::di
     return tasks_to_create;
 }
 
+future<> view_building_worker::register_staging_sstable(sstables::shared_sstable sst, lw_shared_ptr<replica::table> table) {
+    auto table_id = table->schema()->id();
+    auto& tablet_map = _db.get_token_metadata().tablets().get_tablet_map(table_id);
+
+    auto my_host_id = _db.get_token_metadata().get_topology().my_host_id();
+    auto shard = get_sstable_shard_id(*sst);
+    auto tid = get_sstable_tablet_id(tablet_map, *sst);
+    auto last_token = tablet_map.get_last_token(tid);
+
+    // Task id doesn't matter, it'll be created by save_view_building_tasks()
+    service::view_building::view_building_task task {
+        utils::UUID{}, service::view_building::view_building_task::task_type::process_staging, service::view_building::view_building_task::task_state::idle,
+        table_id, ::table_id{}, {my_host_id, shard}, last_token
+    };
+    _staging_sstables[table_id][last_token].push_back(std::move(sst));
+
+    co_await container().invoke_on(0, [task = std::move(task)] (auto& vbw) -> future<> {
+        auto guard = co_await vbw._group0_client.start_operation(vbw._as, service::raft_timeout{});
+        co_await vbw.save_view_building_tasks(std::move(guard), {std::move(task)});
+    });
```

Comment:
I think it is a mistake to do a group0 operation here. The view build worker should be notified about the new sstables and then it should, itself, work on adding the new view building tasks. A group0 operation might fail which can cause the sstable upload to fail - this is completely unnecessary.

Again, this should be fixed ASAP, but can be done in a follow-up.

Reply:
fixed